hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java

```diff
@@ -220,6 +220,24 @@ default Stream<R> stream() {
 	@Incubating
 	long getResultCount();
 
+	/**
+	 * Execute the query and return the results for the given
+	 * {@linkplain KeyedPage page}, using key-based pagination.
+	 *
+	 * @param page the key-based specification of the page as
+	 *        an instance of {@link KeyedPage}
+	 *
+	 * @return the query results and the key of the next page
+	 *         as an instance of {@link KeyedResultList}
+	 *
+	 * @since 6.5
+	 *
+	 * @see KeyedPage
+	 * @see KeyedResultList
+	 */
+	@Incubating
+	KeyedResultList<R> getKeyedResultList(KeyedPage<R> page);
```

Comment:
hey @yrodiere I was looking into something else against ORM 6.5 and noticed this change. This will require changes to  `HibernateOrmSearchQueryAdapter` and this time we can't just add the method ahead of time, since `KeyedResultList` and `KeyedPage` aren't currently available in 6.4 🙈

Reply:
Ah OK, no problem, just add a default implementation somewhere which throws unsupported.

Do you implement that interface directly, or one of the abstract classes?

Reply:
hey Gavin! 😃 
We are extending the `org.hibernate.query.spi.AbstractQuery`. can we add
```java
	@Override
	public KeyedResultList<R> getKeyedResultList(KeyedPage<R> keyedPage) {
		throw new UnsupportedOperationException("Getting keyed result list is not supported by this query.");
	}
```
to it?

Reply:
Yes of course.

Reply:
Here's a PR https://github.com/hibernate/hibernate-orm/pull/7938